packages/flutter/lib/src/foundation/change_notifier.dart

```diff
@@ -119,7 +119,24 @@ class ChangeNotifier implements Listenable {
   int _reentrantlyRemovedListeners = 0;
   bool _debugDisposed = false;
 
-  bool _debugAssertNotDisposed() {
+  /// Used by subclasses to assert that the [ChangeNotifier] has not yet been
+  /// disposed.
+  ///
+  /// {@tool snippet}
+  /// The `debugAssertNotDisposed` function should only be called inside of an
+  /// assert, as in this example.
+  ///
+  /// ```dart
+  /// class MyNotifier with ChangeNotifier {
+  ///   void doUpdate() {
+  ///     assert(debugAssertNotDisposed());
+  ///     // ...
+  ///   }
+  /// }
+  /// ```
+  /// {@end-tool}
+  @protected
+  bool debugAssertNotDisposed() {
```

Comment:
This is a breaking change, breaking all classes that implement `ChangeNotifier`.

i.e. I'm using an [alternative `ChangeNotifier` implementation](https://github.com/wiredashio/wiredash-sdk/blob/stable/lib/src/utils/changenotifier2.dart) that doesn't throw when disposed multiple times.

It could be implemented as mixin instead